File: perl/emit_log_topic.pl
#!/usr/bin/perl

use strict;
use warnings;

$|++;
use AnyEvent;
use Net::RabbitFoot;

my $conn = Net::RabbitFoot->new()->load_xml_spec()->connect(
    host => 'localhost',
    port => 5672,
    user => 'guest',
    pass => 'guest',
    vhost => '/',
);

my $channel = $conn->open_channel();

$channel->declare_exchange(
    exchange => 'topic_logs',
    type => 'topic',
);

my $routing_key = $ARGV[0] || 'anonymous.info';
my $msg = join(' ', @ARGV[1..$#ARGV]) || 'Hello World!';

$channel->publish(
    exchange => 'topic_logs',
    routing_key => $routing_key,
    body => $msg,
);

print " [x] Sent $routing_key:$msg\n";

$conn->close();

File: dart/emit_log_topic.dart
import "package:dart_amqp/dart_amqp.dart";

void main(List<String> arguments) {
    ConnectionSettings settings = new ConnectionSettings(
        host: "localhost"
    );

    Client client = new Client(settings: settings);

    String routingKey = arguments.length < 1 ? "anonymous.info" : arguments[0];
    String msg = arguments.length < 2 ? "Hello World!" : arguments[1];

    client
        .channel()
        .then((Channel channel) =>
                channel.exchange("topic_logs", ExchangeType.TOPIC,
                        durable: false))
        .then((Exchange exchange) {
            exchange.publish(msg, routingKey);
            print(" [x] Sent ${routingKey}: ${msg}");
            return client.close();
        });
}
